Exercise_4.java

```diff
@@ -32,7 +37,29 @@ static void inorder(Node temp)
     /*function to insert element in binary tree */
     static void insert(Node temp, int key) 
     { 
-
+        if(temp==null) 
+        temp=new Node(key) ;
+        else{
+        Queue<Node> q=new LinkedList<>();
+        q.add(temp);
+        while(!q.isEmpty()){
+           Node n=q.poll();

```

Comment:
Why didn't we use size variable here. We use size variable in BFS problems and for some we don't